### PR TITLE
2) React: Product Listing Page

### PR DESCRIPTION
**Pull request by varun13169**
## IGNORE: React Files, only added in this commit

Add Product Listing Page in react.

image

### CODE REVIEW

File: siteComponents/utils.css [hunk 1]
```diff
@@ -1,3 +1,86 @@
...
+    width: 100%;
+    height: 100%;
+    padding: 2rem;
+    display: grid;
+}
+
+.dui-card-prod-hzntl__img {
+    object-fit: cover;
+    width: 100%;
+    height: 100%;
```
Comment (MayankKumar10) [inline siteComponents/utils.css]: height by default is 100%

File: pages/productListingPage/productListingPage-vert.css [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    width: 100%;
+    height: 7rem;
+    background-color: var(--dui-light-theme-bg);
+    box-shadow: 0px 0px 0px 2px var(--dui-primary-color-p2-lgt-35pct);
+
+    position: fixed;
+    bottom: 0;
+    left: 0;
+
+    display: flex;
```
Comment (MayankKumar10) [inline pages/productListingPage/productListingPage-vert.css]: make helper class for flex

**Review by MayankKumar10 — APPROVED** [review]
Look into these Suggestions everything else Looks good to me you can merge.

File: pages/productListingPage/productListingPage-vert.css (hunk 2 above)
Comment (varun13169) [reply pages/productListingPage/productListingPage-vert.css]: I have made a choice of not using class for single CSS property.

File: siteComponents/utils.css (hunk 1 above)
Comment (varun13169) [reply siteComponents/utils.css]: As far as I know, height by default is fit-content.

File: src/styles.css [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+/* footer */
+.page-footer {
+    grid-area: p_footer;
+}
+
+@media only screen and (max-width: 855px) {
+    .page-wrap {
+    grid-auto-rows: 15rem auto auto 7.5rem;
+    }
+}
```
Comment (deekshithmd) [inline src/styles.css:51]: couldn't find much mistakes or errors, as of now looks good

**Review by deekshithmd — APPROVED** [review]

File: public/index.html [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	<!--
+      This HTML file is a template.
+      If you open it directly in the browser, you will see an empty page.
+
+      You can add webfonts, meta tags, or analytics to this file.
+      The build step will place the bundled scripts into the <body> tag.
+
+      To begin the development, run `npm start` or `yarn start`.
+      To create a production bundle, use `npm run build` or `yarn build`.
+    -->
```
Comment (zoef-yt) [inline public/index.html:40]: remove commented code

File: src/styles.css [hunk 5]
```diff
@@ -0,0 +1,51 @@
+@import url("https://delta-ui-preview-2.netlify.app/library/component.css");
+
+/* @import "siteComponents/sidebar.css";
+@import "siteComponents/footer.css";
+@import "siteComponents/utils.css"; */
```
Comment (zoef-yt) [inline src/styles.css:5]: remove the comment code.

File: src/pages/productListingPage/productListingPage.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                <div className="dui-util-clr-prim-p2">
+                    <svg xmlns="http://www.w3.org/2000/svg" className="icon icon-tabler icon-tabler-filter" width="56"
+                        height="56" viewBox="0 0 24 24" stroke-width="1.5" stroke="#9A9A9A" fill="none"
+                        stroke-linecap="round" stroke-linejoin="round">
+                        <path stroke="none" d="M0 0h24v24H0z" fill="none" />
+                        <path d="M5.5 5h13a1 1 0 0 1 .5 1.5l-5 5.5l0 7l-4 -3l0 -4l-5 -5.5a1 1 0 0 1 .5 -1.5" />
+                    </svg>
+                    <p className="dui-util-fw-bld">Filter</p>
+                </div>
+            </button> */}
```
Comment (zoef-yt) [inline src/pages/productListingPage/productListingPage.jsx:35]: remove commented code.

File: src/pages/productListingPage/productListingPage.jsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+      </section>
+
+      <section className="page-sidebar">
+        <Sidebar></Sidebar>
+      </section>
+
+      <section className="page-main">
+        <main className="main-content dui-util-spc-pad-2_4rem-m">
+          {[1,2,3,4,5,6,7,8].map((itemDetails) => {
+            // console.log(a);
```
Comment (zoef-yt) [inline src/pages/productListingPage/productListingPage.jsx:18]: remove commented code.

File: src/components/Sidebar/Sidebar.jsx [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+          Category
+        </p>
+
+        <div>
+          {/* <!-- Checkbox Component Starts --> */}
+          <label className="dui-inp-chkbox dui-util-txt-sm">
+            Indoor
+            <input type="checkbox" defaultChecked />
+            <span className="dui-inp-chkbox__checkmark"></span>
+          </label>
```
Comment (zoef-yt) [inline src/components/Sidebar/Sidebar.jsx:42]: These can be converted into a component as you are using them again and again.

File: src/components/Card/Card.jsx [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    });
+  const secAction =
+    props.secAction ??
+    (() => {
+      console.log("No Seconday Action Defined");
+    });
+  const itemDetails = props.itemDetails ?? {name: "Amazing Pot"};
+  const itemName = itemDetails.name;
+  console.log("Inside card: ");
+  console.log(props);
```
Comment (zoef-yt) [inline src/components/Card/Card.jsx:17]: Remove all the console.logs.

**Review by zoef-yt — CHANGES_REQUESTED** [review]

File: src/styles.css (hunk 3 above)
Comment (varun13169) [reply src/styles.css:51]: Thanks.

File: public/index.html (hunk 4 above)
Comment (varun13169) [reply public/index.html:40]: will be addressing the same in upcoming refactoring.

File: src/styles.css (hunk 5 above)
Comment (varun13169) [reply src/styles.css:5]: will be removed in upcoming refactoring.

File: src/pages/productListingPage/productListingPage.jsx (hunk 6 above)
Comment (varun13169) [reply src/pages/productListingPage/productListingPage.jsx:35]: will be removed in upcoming refactoring.

File: src/components/Sidebar/Sidebar.jsx (hunk 8 above)
Comment (varun13169) [reply src/components/Sidebar/Sidebar.jsx:42]: will consider in upcoming refactoring.

File: src/components/Card/Card.jsx (hunk 9 above)
Comment (varun13169) [reply src/components/Card/Card.jsx:17]: will be removed in upcoming refactoring.

File: src/pages/productListingPage/productListingPage.jsx (hunk 7 above)
Comment (varun13169) [reply src/pages/productListingPage/productListingPage.jsx:18]: will be removed in upcoming refactoring.

**Pull request merged**